workers.tf

```diff
@@ -143,7 +143,10 @@ resource "aws_launch_configuration" "workers" {
   image_id = lookup(
     var.worker_groups[count.index],
     "ami_id",
-    lookup(var.worker_groups[count.index], "platform", local.workers_group_defaults["platform"]) == "windows" ? local.default_ami_id_windows : local.default_ami_id_linux,
+    lookup(
+      local.workers_group_defaults, "ami_id",
+      lookup(var.worker_groups[count.index], "platform", local.workers_group_defaults["platform"]) == "windows" ? local.default_ami_id_windows : local.default_ami_id_linux
+    )
```

Comment:
I would prefer something easier to read. Something like this [here](https://github.com/terraform-aws-modules/terraform-aws-eks/blob/908c656cf5da50d576d77696efac1d76dfb86a07/local.tf#L22-L23) :

```hcl
default_ami_id_linux   = lookup(local.workers_group_defaults, "ami_id", data.aws_ami.eks_worker.id)
default_ami_id_windows = lookup(local.workers_group_defaults, "ami_id", data.aws_ami.eks_worker_windows.id)
```
And leave workers.tf and workers_launch_template.tf unchanged. What do you think ?

Reply:
Yeah, I see where you coming from but then the same holds true for worker_launch_template specific overrides... I'd rather have it where it is being used so that when people look at the code and follow the logic they'll see the order of precendence... make sense?

Reply:
Okay, I'll implement it this way, check back in a sec.

Reply:
@barryib as you can see I have implemented your suggestion. One thing that worries me now is that in the plan it doesn't show the default AMI IDs but rather `null` so I'll spend some more time on this.

Reply:
Okay, implemented as you suggested. Instead of `local.workers_group_defaults` I used `var.workers_group_defaults` however.